internal/cli/api.go

```diff
@@ -0,0 +1,186 @@
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/auth0/auth0-cli/internal/ansi"
+)
+
+const apiDocsURL = "https://auth0.com/docs/api/management/v2"
+
+var apiFlags = apiCmdFlags{
+	Data: Flag{
+		Name:         "RawData",
+		LongForm:     "data",
+		ShortForm:    "d",
+		Help:         "JSON data payload to send with the request.",
+		IsRequired:   false,
+		AlwaysPrompt: false,
+	},
+}
+
+var apiValidMethods = map[string]bool{
+	"GET":    true,
+	"POST":   true,
+	"PUT":    true,
+	"PATCH":  true,
+	"DELETE": true,
+}
+
+type (
+	apiCmdFlags struct {
+		Data Flag
+	}
+
+	apiCmdInputs struct {
+		RawMethod string
+		RawURI    string
+		RawData   string
+		Method    string
+		URL       *url.URL
+		Data      io.Reader
+	}
+)
+
+func apiCmd(cli *cli) *cobra.Command {
+	var inputs apiCmdInputs
+
+	cmd := &cobra.Command{
+		Use:   "api <method> <uri>",
+		Args:  cobra.RangeArgs(1, 2),
+		Short: "Makes an authenticated HTTP request to the Auth0 Management API",
+		Long: fmt.Sprintf(
+			`Makes an authenticated HTTP request to the Auth0 Management API and prints the response as JSON.
+
+The method argument is optional, and when you don’t specify it, the command defaults to GET for requests without data
+and POST for requests with data.
+
+%s  %s
+
+%s  %s`,
+			"Auth0 Management API Docs:\n", apiDocsURL,
+			"Available Methods:\n", "GET, POST, PUT, PATCH, DELETE",
+		),
+		Example: `auth0 api "/organizations?include_totals=true"
+auth0 api get "/organizations?include_totals=true"
+auth0 api clients --data "{\"name\":\"apiTest\"}"
+`,
+		RunE: apiCmdRun(cli, &inputs),
+	}
+
+	apiFlags.Data.RegisterString(cmd, &inputs.RawData, "")
+
+	return cmd
+}
+
+func apiCmdRun(cli *cli, inputs *apiCmdInputs) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		if err := inputs.fromArgs(args, cli.tenant); err != nil {
+			return fmt.Errorf("failed to parse command inputs: %w", err)
```

Comment:
Per [CLI guidelines](https://clig.dev/#errors), this error could be more human friendly. Granted, we will not know the exact errors that arise, but we can make some educated guesses and provide actionable guidance to the end user to rectify.

Reply:
The returned error wraps all inner errors so the final one ends up to be for example:

```
$ auth0 api z z 

=== tenat-example.auth0.com error

 ▸    failed to parse command inputs: invalid method given: Z, accepting only GET, POST, PUT, PATCH, DELETE
exit status 1
```

What do you suggest so we make it more human friendly?